onnxruntime/core/session/inference_session.h

```diff
@@ -863,6 +863,14 @@ class InferenceSession {
       cached_execution_provider_for_graph_replay_ = execution_provider;
     }
 
+    void SetGraphAnnotation(int graph_annotation_id) {
+      graph_annotation_id_ = graph_annotation_id;
+      if (!cached_execution_provider_for_graph_replay_) {
+        ORT_THROW("Cached EP instance for graph replay is not set yet before calling SetGraphAnnotation()");
```

Comment:
In what situation that this will be hit?
Move this to the beginning of this function so that `graph_annotation_id_` is not updated when things is wrong.

Reply:
SetGraphAnnotation() is removed